apps/stripe/src/app/api/saleor/transaction-initialize-session/use-case.ts

```diff
@@ -24,6 +27,64 @@ export class TransactionInitializeSessionUseCase {
     this.stripePaymentIntentsApiFactory = deps.stripePaymentIntentsApiFactory;
   }
 
+  private prepareStripeCreatePaymentIntentParams(
+    event: TransactionInitializeSessionEventFragment,
+  ): Stripe.PaymentIntentCreateParams {
+    const stripeMoneyResult = StripeMoney.createFromSaleorAmount({
+      amount: event.action.amount,
+      currency: event.action.currency,
+    });
+
+    if (stripeMoneyResult.isErr()) {
+      this.logger.error(
+        "Failed to convert amount / currency coming from Saleor to one used by Stripe - throwing",
+        {
+          error: stripeMoneyResult.error,
+        },
+      );
+
+      throw new TransactionInitializeSessionUseCase.UseCaseError("Failed to create Stripe money", {
+        cause: stripeMoneyResult.error,
+      });
+    }
+
+    return {
+      amount: stripeMoneyResult.value.getAmount(),
+      currency: stripeMoneyResult.value.getCurrency(),
+    };
+  }
+
+  private prepareResponseToSaleor(
+    stripePaymentIntentResponse: Stripe.PaymentIntent,
+  ): ReturnType<typeof buildSyncWebhookResponsePayload<"TRANSACTION_INITIALIZE_SESSION">> {
+    const saleorMoneyResult = SaleorMoney.createFromStripe({
+      amount: stripePaymentIntentResponse.amount,
+      currency: stripePaymentIntentResponse.currency,
+    });
+
+    if (saleorMoneyResult.isErr()) {
+      this.logger.error(
+        "Failed to convert amount / currency coming from Stripe to one used by Saleor - throwing",
+        {
+          error: saleorMoneyResult.error,
+        },
+      );
+
+      throw new TransactionInitializeSessionUseCase.UseCaseError("Failed to create Saleor money", {
+        cause: saleorMoneyResult.error,
+      });
+    }
+
+    return {
+      result: "CHARGE_REQUESTED",
+      amount: saleorMoneyResult.value.getAmount(),
+      pspReference: stripePaymentIntentResponse.id,
+      data: {
+        stripeClientSecret: stripePaymentIntentResponse.client_secret,
+      },
```

Comment:
lets wrap that to be type safe

maybe extend type from app-sdk and declare our type that types `data`?

Reply:
Fixed in https://github.com/saleor/apps/pull/1851/commits/aad47a2d9735da9e92e41769b811aa4ada588ba1